apps/chat/channels.py

```diff
@@ -103,6 +103,8 @@ def initialize(self):
 
     @property
     def chat_id(self) -> int:
+        if self.experiment_session and self.experiment_session.external_chat_id:
+            return self.experiment_session.external_chat_id
```

Comment:
what's the advantage of doing this?

Reply:
So we can instantiate a channel (TelegramChannel, WhatsappChannel etc) in two ways:
1. WhatsappChannel(...)
2. ChannelBase.from_experiment_session(experiment_session)

Consider the second case: When the channel is freshly instantiated, `channel.message` will be `None`, so if we try to send a message to the user, then calling `channel.chat_id` will cause an error, because it calls `get_chat_id_from_message`, which tries to get the `chat_id` from the user's message, but we don't have that. We do have the experiment session which have this info though, so we might as well look there first.

You might be wondering "what about the first case then?". It is perfectly valid to do something like `WhatsappChannel(experiment_channel=experiment_channel)` where `experiment_channel` will be without an experiment session. This is in fact the case when a user first sends a message. In this case, we'll fall back to the user message to get the chat id.

If we want to send a message to the user and use `WhatsappChannel(experiment_channel=experiment_channel)`, then it will fail, since then it wouldn't be able to determine the `chat_id`. We then have to pass in an `experiment_session` instance as well.

I hope I explained this correctly.

P.S. The reason why this is change is made only now is because the `new_bot_message` method was implemented by each channel and it looked at `self.experiment_session.external_chat_id` to determine the `chat_id`.